### PR TITLE
feat(api-client): better server handling

### PR DESCRIPTION
**Pull request by hanspagel**
WIP

Need to fix tests

**Problem**

The server logic in the reference covered some edge cases that we didn't cover in the client.

**Solution**

This PR:
* moves the server logic to a separate function
* add tests
* covers some more edge cases
* adds baseServerURL to the documentation
* improves the Zod schema for the servers
  * closer to the specification
  * removes `'default'` as the default value for variables
  * if `enum` is defined: enforces `default` is a possible value
  * if a single server is invalid, it's omitted, but the others servers are still there
  * the data is validated against the schema first *and then* modified, this makes it more robust
  * updated the comments

Screenshot 2025-01-21 at 17 02 16

**Checklist**

I’ve gone through the following:

- [x] I’ve added an explanation _why_ this change is needed.
- [x] I’ve added a changeset (`pnpm changeset`).
- [x] I’ve added tests for the regression or new feature.
- [x] I’ve updated the documentation.

### CODE REVIEW

File: packages/oas-utils/src/transforms/import-spec.ts [hunk 1]
```diff
@@ -478,3 +455,57 @@ export async function importSpecToWorkspace(
...
+): Server[] {
+  if (!servers || !Array.isArray(servers)) return []
+
+  return servers
+    .map((server): Server | undefined => {
+      try {
+        // Validate the server against the schema
+        const parsedSchema = serverSchema.parse(server)
+
+        console.log()
```
Comment (amritk) [inline packages/oas-utils/src/transforms/import-spec.ts:474]:
```suggestion
```

**Review by amritk — APPROVED** [review]
Has some test and some type issues to resolve but, love the code! We should continue pulling logic out and testing it!!! One question about the OpenAPIV3_1 type is it already de-referenced? Because we have de-referenced already but the type thinks there could be a reference still, or am I mistaken.

**Comment by hanspagel** [comment]
Those types are usually for the original document with unresolved references, but servers and variables don’t allow references, so there’s no difference. Not super happy with that solution, I don't know. Just wanted to get rid of another random custom type.
